### PR TITLE
Replace stringlist in analysis_config

### LINKED ISSUE #2493: Replace `stringlist_type` with `std::vector<string>` in analysis_config

### PR DESCRIPTION
**Pull request by dafeda**
**Issue**
Resolves #2493

### CODE REVIEW

File: libres/lib/enkf/analysis_config.cpp [hunk 1]
```diff
@@ -154,12 +154,12 @@ int analysis_config_get_min_realisations(const analysis_config_type *config) {
     return config->min_realisations;
 }
 
-stringlist_type *
+std::vector<std::string>
 analysis_config_alloc_module_names(const analysis_config_type *config) {
```
Comment (joakim-hove) [inline libres/lib/enkf/analysis_config.cpp:158]: It has been a strong pattern: Functions with `_alloc_` in the name allocate memory and a corresponding `xxx_free` needs to be somewhere, with `std::vector<>` and value semantics I would suggest renaming the function e.g. `analysis_config_module_names()`

File: res/enkf/analysis_config.py [hunk 2]
```diff
@@ -245,9 +243,8 @@ def activeModuleName(self):
         """:rtype: str"""
         return self._get_active_module_name()
 
-    def getModuleList(self) -> StringList:
-        """:rtype: StringList"""
-        return self._get_module_list()
+    def getModuleList(self):
```
Comment (oyvindeide) [inline res/enkf/analysis_config.py:246]: Can you add the type hint back?

**Review by oyvindeide — APPROVED** [review]
LGTM! Just a minor comment

File: res/enkf/analysis_config.py (hunk 2 above)
Comment (oyvindeide) [reply res/enkf/analysis_config.py:246]: Can you also fix the places using this call? They are explicitly casting to list, that is no longer required.

Comment (dafeda) [reply res/enkf/analysis_config.py]: Thanks!

**Pull request merged**